### ISSUE TITLE
When using .NotNull for input validator, client side proxies also disallow empty string.

### ISSUE THREAD

**Issue by bnordli** [dolittle/Bifrost#671]
After fixing #670, a lot of previously dynamic input validators got client side proxies. Most concept rules (probably all of them should) use .NotNull() validator. This is transformed to the "required" rule:

``` javascript
  required: Bifrost.validation.Rule.extend(function () {
        this.validate = function (value) {
            return !(Bifrost.isUndefined(value) || Bifrost.isNull(value) || value === "");
        };
    })
```

which also fails when value is empty! This should not be the intended behavior.

(The problem is with RequiredGenerator, which also claims it generates from INotNullValidator. There should be a separate NotNullGenerator.)

**Comment by bnordli** [comment]
This is fixed in the ProCoSys fork.

**Comment by bnordli** [comment]
This was actually fixed in #672.
